claim/claim.c

```diff
@@ -109,12 +109,26 @@ void claim_agent(char *claiming_arguments)
 #endif
 }
 
+extern int aclk_connected, aclk_kill_link;
+extern void aclk_graceful_disconnect();
+/* Change the claimed state of the agent.
+ *
+ * This only happens when the user has explicitly requested it:
+ *   - via the cli tool by reloading the claiming state
+ *   - after spawning the claim because of a command-line argument
+ * If this happens with the ACLK active under an old claim then we MUST KILL THE LINK
+ */
 void load_claiming_state(void)
 {
     if (claimed_id != NULL) {
         freez(claimed_id);
         claimed_id = NULL;
```

Comment:
Very very small window that might cause a crash due to `char *agent_id = is_agent_claimed();` in `aclk_get_challenge` (L1242) if called in between the freez and the set to null.

Reply:
This is true and I've been wondering about it recently. The risk is minor but it is possible that accessing the cli-tool could trigger it.